STYLEGUIDE.md

```diff
@@ -0,0 +1,230 @@
+:---|---|---|---|---|---|---:|
+[System](EXAMPLE) | | | | | | 10000 |
+[System.Xml](EXAMPLE)|
+[TaleWorlds.SaveSystem](EXAMPLE) |
+|
+</span>
+
+> Omit elements entirely if not relavent to that class (eg: parent class)
+
+> For the non-savable classes, use ```'color:red'``` and change to text to "Not Savable".
+
+After this class overview, a small code snippet should exist showing how to use / initialize the class. **This should not be exhaustive**, just a quick reference.
+```csharp
+new InformationMessage("Hello World!")
+```
+
+The above would be a valid snippet for the ```InformationMessage``` Class, but can be longer if the extra length is used to give important context about the snippet.
+
+Directly after this intro snippet, we have a section outlining the variables of the class. All variables are assumed to have private setters unless otherwise noted. Each variable should contain all of the following information:
+
+* Name
+* Type (with link to that object or objects)
+* Notes
+
+## Variables
+
+<span style= "font-size:12px">
+
+| Name | Type | Notes |
+| --- | --- | --- |
+| HeadArmor | int |
+| MeshesMask | [SkinMask](EXAMPLE) |
+| BeardCoverType | [ArmorComponent](EXAMPLE).[BeardCoverTypes](EXAMPLE)
+| ReinsRopeMesh | string | returns ```this.ReinsMesh + "_rope";```|
+|...|...|...|
+|
+</span>
+
+> Always add a blank row ( | ) at the bottom of tables to square them off. Note that this table follows the same font size rules as as the class table.
+
+Methods should follow parameters, seperated by a H1 header.
+
+## Methods
+
+<span style= "font-size:12px">
+
+| Type | Method Name | # of Params |
+| --- | --- | --- | --- | --- |
+| void | [SetItem] | 2 |
+| [Equipment](EXAMPLE) | [Clone](EXAMPLE) | 1 | 
+| void | [FillFrom](EXAMPLE) | 2 |
+| ... | ... | ... |
+|
+</span>
+
+
+> The class page should only list out the methods like a table of contents, the detail should be saved for the methods own README page.
+
+Methods should be like a miniature version of a class page, with a small table at the top for parameters (if any), return type, description, etc. as shown below:
+
+<span style= "font-size:12px">
+
+### [Parent Object Name](EXAMPLE)
+## Method Name
+
+| Parameters | Type | default | Optional? |
+|---|---|---|---|
+| Param Name | string | 'Hello World' | ✖ |
+| Param Name | [XmlNode](EXAMPLE) | N/A | ☑ |
+| ... | ... |
+|
+</span>
+
+> Parameters should be in the same order as they appear in the method. Types should also link to their class. Copy paste symbols as needed. You don't need to link to base types (like int or string).
+
+### Detailed description of the method.
+ ___
+ Code Snippets (extended)
+```
+Here is how you do the thing!
+```
+```
+Here is another way to do the thing
+```
+
+___
+
+# Enums
+
+Enums should be placed in their own file at the Class level (on their own page under a package) with the following format:
```

Comment:
I was basing off of the decompiled dlls. Enums are defined in their and don't have anything else in them afaik:

public enum WeaponClass
    {
        // Token: 0x040005BF RID: 1471
        Undefined,
        // Token: 0x040005C0 RID: 1472
        Dagger,
...

Reply:
I am working on Hero class. And it define structs and enums in self.
Where would be good place to document them?

Reply:
If they are unique / private to the class they should probably go somewhere within the class, if they live outside the class then they should go as their own page in the Enums section. Open to recommendations for the former.

Reply:
Ok. I will place it to class on my case.
But according to styleguides enum should be separated from class.
So that lack on develop side :D (also with public variables)